app/http.php

```diff
@@ -0,0 +1,1039 @@
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->json($container);
+    });
+
+
+App::get('/v1/runtimes')
+    ->desc("List currently active runtimes")
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(function (Table $activeRuntimes, Response $response) {
+        $runtimes = [];
+
+        foreach ($activeRuntimes as $runtime) {
+            $runtimes[] = $runtime;
+        }
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_OK)
+            ->json($runtimes);
+    });
+
+App::get('/v1/runtimes/:runtimeId')
+    ->desc("Get a runtime by its ID")
+    ->param('runtimeId', '', new Text(64), 'Runtime unique ID.')
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(function (string $runtimeId, Table $activeRuntimes, Response $response) {
+        $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
+
+        if (!$activeRuntimes->exists($activeRuntimeId)) {
+            throw new Exception('Runtime not found', 404);
+        }
+
+        $runtime = $activeRuntimes->get($activeRuntimeId);
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_OK)
+            ->json($runtime);
+    });
+
+App::delete('/v1/runtimes/:runtimeId')
+    ->desc('Delete a runtime')
+    ->param('runtimeId', '', new Text(64), 'Runtime unique ID.', false)
+    ->inject('orchestrationPool')
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(function (string $runtimeId, Pool $orchestrationPool, Table $activeRuntimes, Response $response) {
+        $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
+        $runtimeId = System::getHostname() . '-' . $runtimeId; // Used in Docker (name)
+
+        if (!$activeRuntimes->exists($activeRuntimeId)) {
+            throw new Exception('Runtime not found', 404);
+        }
+
+        Console::info('Deleting runtime: ' . $runtimeId);
+
+        try {
+            $connection = $orchestrationPool->pop();
+            $orchestration = $connection->getResource();
+            $orchestration->remove($runtimeId, true);
+            $activeRuntimes->del($activeRuntimeId);
+            Console::success('Removed runtime container: ' . $runtimeId);
+        } finally {
+            isset($connection) && $orchestrationPool->push($connection);
+        }
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_OK)
+            ->send();
+    });
+
+App::post('/v1/execution')
+    ->desc('Create an execution')
+    // Execution-related
+    ->param('runtimeId', '', new Text(64), 'The runtimeID to execute.')
+    ->param('payload', '', new Text(8192), 'Data to be forwarded to the function, this is user specified.', true)
+    ->param('variables', [], new Assoc(), 'Environment variables required for the build and execution.', true)
+    ->param('timeout', 15, new Range(1, (int) App::getEnv('OPEN_RUNTIMES_EXECUTOR_MAX_TIMEOUT', "900")), 'Function maximum execution time in seconds.', true)
```

Comment:
```suggestion
    ->param('timeout', 15, new Range(1, (int) App::getEnv('OPEN_RUNTIMES_EXECUTOR_TIMEOUT', "900")), 'Function maximum execution time in seconds.', true)
```

Do we really need MAX in the name ? Doesn't TIMEOUT convey the meaning clear enough?

Reply:
We use MAX because its validator maximum setting - it does not represent actual build timeout time.
Maybe `OPEN_RUNTIMES_EXECUTOR_TIMEOUT_LIMIT` instead?

Reply:
I agree with Christy here. As long as we don't have other uses we don't need to optimize for using TIMEOUT as a namespace.

Reply:
With Eldad we decided to use `OPEN_RUNTIMES_EXECUTOR_EXECUTION_TIMEOUT_LIMIT` in order to make it verbose:

- This is limit, not default timeout
- This is execution timouut, not executor HTTP server timout

Reply:
This is a super long variable name. We should minimise the amount of words.

This variable is built out of 6 different nouns. 4 of them don't add any additional meaning to the variable
OPEN
RUNTIMES
EXECUTOR
LIMIT

If this variable were to be renamed to `EXECUTION_TIMEOUT`, it would still have the same meaning.

Reply:
We are already in the OPR namespace. There is no additional need to use `OPEN_RUNTIMES` in each variable name. 
If namespace is really required, `EXECUTOR_` can be used